chia/util/misc.py

```diff
@@ -165,3 +170,75 @@ def to_batches(to_split: Collection[T], batch_size: int) -> Iterator[Batch[T]]:
             yield Batch(total_size - processed, entries)
     else:
         raise ValueError(f"to_batches: Unsupported type {type(to_split)}")
+
+
+class Handler(Protocol):
+    def __call__(self, signal_number: int, stack_frame: Optional[FrameType], loop: asyncio.AbstractEventLoop) -> None:
+        ...
+
+
+class AsyncHandler(Protocol):
+    async def __call__(
+        self, signal_number: int, stack_frame: Optional[FrameType], loop: asyncio.AbstractEventLoop
+    ) -> None:
+        ...
+
+
+def setup_sync_signal_handler(handler: Handler) -> None:
+    loop = asyncio.get_event_loop()
+
+    if sys.platform == "win32" or sys.platform == "cygwin":
+        for signal_ in [signal.SIGBREAK, signal.SIGINT, signal.SIGTERM]:
+            signal.signal(signal_, functools.partial(handler, loop=loop))
+    else:
+        for signal_ in [signal.SIGINT, signal.SIGTERM]:
+            loop.add_signal_handler(
+                signal_,
+                functools.partial(handler, signal_number=signal_, stack_frame=None, loop=loop),
+            )
+
+
+# signals are a global thing so...  global task reference maintenance is "ok" i guess
+signal_handler_tasks: List[asyncio.Task[None]] = []
```

Comment:
I would have thought the point of having this list would be to join them when shutting down, but I don't see that anywhere

Reply:
https://github.com/Chia-Network/chia-blockchain/commit/e020ff2e191e2d3b37587c948214b6d31cfc73e6 I think there's more refactor that could be worthwhile around this, but maybe this is an ok layer to stop at for this PR.